BitwardenShared/Core/Auth/Repositories/AuthRepository.swift

```diff
@@ -520,8 +520,10 @@ extension DefaultAuthRepository: AuthRepository {
             )
         )
 
+        // Get the user's ID before it's removed from `StateService`.
+        let userId = try await stateService.getActiveAccountId()
         try await stateService.deleteAccount()
-        await vaultTimeoutService.remove(userId: nil)
+        await vaultTimeoutService.remove(userId: userId)
```

Comment:
Passing `nil` as the `userId` here was causing it to use the active account. But the active account changes after the account is deleted. Passing in the user ID before it's removed fixes that.

Reply:
Nice catch 🎉